File: campagne/gestione-case-vacanza-proprietari/instagram-ready/render.mjs
import pw from '/opt/node22/lib/node_modules/playwright/index.js';
const { chromium } = pw;
import path from 'node:path';
import { fileURLToPath, pathToFileURL } from 'node:url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const src = path.join(__dirname, 'src');
const out = path.join(__dirname, 'out');

const jobs = [
  ['storia-1.html', 1080, 1920],
  ['storia-2.html', 1080, 1920],
  ['storia-3.html', 1080, 1920],
  ['storia-4.html', 1080, 1920],
  ['storia-5.html', 1080, 1920],
  ['carosello-1.html', 1080, 1350],
  ['carosello-2.html', 1080, 1350],
  ['carosello-3.html', 1080, 1350],
  ['carosello-4.html', 1080, 1350],
  ['carosello-5.html', 1080, 1350],
  ['post-prova.html', 1080, 1350],
  ['reel-cover.html', 1080, 1920],
];

const browser = await chromium.launch({
  executablePath: process.env.PW_CHROME || undefined,
});
for (const [file, w, h] of jobs) {
  const page = await browser.newPage({ viewport: { width: w, height: h }, deviceScaleFactor: 1 });
  const url = pathToFileURL(path.join(src, file)).href;
  await page.goto(url, { waitUntil: 'load' });
  try { await page.evaluate(() => document.fonts.ready); } catch {}
  await page.waitForTimeout(400);
  const el = await page.$('.frame');
  const outFile = path.join(out, file.replace('.html', '.png'));
  await el.screenshot({ path: outFile });
  console.log('rendered', path.basename(outFile), `${w}x${h}`);
  await page.close();
}
await browser.close();
console.log('done');

File: .claude/skills/giornata/render.py
"""Renderizza le artboard .dc.html di una giornata in PNG.

    python3 .claude/skills/giornata/render.py campagne/giornaliero/<AAAA-MM-GG>

Legge grafiche/canvas.json, apre ogni .dc.html in Chromium alla dimensione
esatta dell'artboard e salva lo screenshot in png/. Serve per i sei controlli
di qualita' di master-template.md, che si fanno sul PNG renderizzato.
"""
import json
import os
import sys

from playwright.sync_api import sync_playwright


def main(giorno):
    grafiche = os.path.join(giorno, 'grafiche')
    png_dir = os.path.join(giorno, 'png')
    os.makedirs(png_dir, exist_ok=True)

    with open(os.path.join(grafiche, 'canvas.json')) as f:
        artboards = json.load(f)['artboards']

    with sync_playwright() as p:
        # Il Chromium preinstallato puo' avere un numero di build diverso da
        # quello che si aspetta la libreria: si punta al binario presente
        # invece di scaricarne un altro.
        import glob
        cands = sorted(glob.glob('/opt/pw-browsers/chromium-*/chrome-linux/chrome'))
        browser = p.chromium.launch(executable_path=cands[-1]) if cands else p.chromium.launch()
        for ab in artboards:
            page = browser.new_page(viewport={'width': ab['w'], 'height': ab['h']},
                                    device_scale_factor=1)
            page.goto('file://' + os.path.abspath(os.path.join(grafiche, ab['file'])))
            page.wait_for_timeout(1200)  # webfont Archivo/Manrope
            out = os.path.join(png_dir, ab['file'].replace('.dc.html', '.png'))
            page.locator('.frame').screenshot(path=out)
            page.close()
            print('%-16s -> %s' % (ab['file'], out))
        browser.close()
    print('\n%d PNG in %s' % (len(artboards), png_dir))


if __name__ == '__main__':
    main(sys.argv[1] if len(sys.argv) > 1 else '.')
